client/services/SolutionHandlerServiceSpec.js

```diff
@@ -0,0 +1,232 @@
+// Copyright 2017 The TIE Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for SolutionHandlerService.
+ */
+
+describe('SolutionHandlerService', function() {
+  var SolutionHandlerService;
+  var TaskObjectFactory;
+  var orderedTasks;
+  var auxiliaryCode;
+
+  beforeEach(module('tie'));
+  beforeEach(inject(function($injector) {
+    SolutionHandlerService = $injector.get(
+      'SolutionHandlerService');
+  }));
+
+  // Mock tasks for preprocessing.
+  beforeEach(inject(function($injector) {
+    TaskObjectFactory = $injector.get(
+      'TaskObjectFactory');
+
+    var taskDict = [
+      {
+        instructions: [''],
+        prerequisiteSkills: [''],
+        acquiredSkills: [''],
+        inputFunctionName: null,
+        outputFunctionName: null,
+        mainFunctionName: 'mockMainFunction',
+        correctnessTests: [{
+          input: 'testcase_0',
```

Comment:
It might be better to name this 'task_1_correctness_test_1' or similar, and check explicitly for the substring 'task_1' in the student code. That would make it easier to read the test cases -- the idea here is to make obvious, to someone who reads only the student code, which test cases are expected to fail. Otherwise lots of careful cross-referencing ends up being needed.

Ditto for the other input strings.